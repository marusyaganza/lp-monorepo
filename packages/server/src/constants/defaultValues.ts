import { Language, Role, SortWordsBy, Tense } from '../generated/graphql';

export const WORDS_PER_PAGE = 5;
export const DEFAULT_LANGUAGE = Language.English;
export const DEFAULT_WORDS_SORT_BY = SortWordsBy.UpdatedAt;
export const DEFAULT_ROLE = Role.Member;
export const DEFAULT_TENSE = Tense.Pind;
export const DEFAULT_TOKEN_TTL = 7;
// to clean db every hour + 10 min just in case
export const DEMO_DB_TTL = 4200;
export const DEFAULT_MONGO_CONNECTION = 'mongodb://localhost:27017';
export const DEFAULT_OPTIONS__SAMPLE_SIZE = 50;
