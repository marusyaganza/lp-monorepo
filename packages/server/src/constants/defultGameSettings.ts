import { Game, Language, SettingsByGame } from '../generated/graphql';

export const DEFAULT_GAMES_SETTINGS: SettingsByGame = {
  [Game.Audio]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.Image]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.Speaking]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.TypeWord]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.SelectDef]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 6,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.SelectWord]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 6,
    timesToLearn: 5,
    languages: [Language.English, Language.Spanish]
  },
  [Game.Conjugation]: {
    wordsPerGame: 2,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.Spanish]
  },
  [Game.Gender]: {
    wordsPerGame: 6,
    minWords: 1,
    optionsPerGame: 0,
    timesToLearn: 5,
    languages: [Language.Spanish]
  }
};
