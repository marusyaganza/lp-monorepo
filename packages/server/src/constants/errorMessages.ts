export const ERROR_MESSAGES = {
  NOT_AUTHENTICATED: 'Login to perform this operation',
  NOT_AUTHORIZED: 'You do not have permission to perform this operation',
  INCORRECT_CREDS: 'email or password is incorrect',
  SIGN_UP_FAILED: 'sign up operation failed',
  SING_UP_UNAVAILABLE:
    'Sign up operation is not available now. Sorry for inconvenience.',
  GAME_NOT_FOUND: 'Game not found',
  GAME_GENERATION_FAILED: 'Game data generation failed'
};
