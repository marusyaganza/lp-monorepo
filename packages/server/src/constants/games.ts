import { Game, GameConfig, Language } from '../generated/graphql';
export const GAMES: GameConfig[] = [
  {
    name: 'WordEcho',
    id: 'audio',
    desc: 'Listen, Type, Master. Sharpen language skills through auditory engagement.',
    type: Game.Audio,
    orderNum: 0,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'LexiSense',
    desc: 'Define & Conquer.Match words with their meanings intuitively.',
    id: 'select_definition',
    type: Game.SelectDef,
    orderNum: 1,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'Definify: Match & Learn',
    desc: 'Associate definitions with the right words. A captivating word puzzle adventure!',
    id: 'select_word',
    type: Game.SelectWord,
    orderNum: 2,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'WordRecall: Define & Type',
    desc: 'Type words matching given definitions. Boost vocabulary in engaging rounds!',
    id: 'type_word',
    type: Game.TypeWord,
    orderNum: 3,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'PictureWord Blitz',
    desc: 'Identify objects from pictures and type their names to enhance vocabulary skills!',
    id: 'image',
    type: Game.Image,
    orderNum: 4,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'VoiceWord Challenge',
    desc: 'Speak words from clues and images. Sharpen your vocabulary and speaking skills fast!',
    id: 'speaking',
    type: Game.Speaking,
    orderNum: 5,
    languages: [Language.English, Language.Spanish]
  },
  {
    name: 'Conjugation Quest',
    desc: 'Master verb conjugations across tenses and languages in fast-paced, interactive challenges!',
    id: 'conjugation',
    type: Game.Conjugation,
    orderNum: 6,
    languages: [Language.Spanish]
  },
  {
    name: 'GenderWord Challenge',
    desc: 'Choose nouns and match their gender. Test linguistic skills in thrilling rounds!',
    id: 'gender',
    type: Game.Gender,
    orderNum: 7,
    languages: [Language.Spanish]
  }
];
