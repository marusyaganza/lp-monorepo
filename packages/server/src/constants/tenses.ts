import { Tense } from '../generated/graphql';

// Uncomment more tenses as needed
export const TENSES: Record<Tense, string> = {
  pind: 'Presente indicativo',
  pprf: 'Pretérito perfecto simple',
  impf: 'Imperativo',
  pret: 'Pretérito imperfecto',
  futr: 'Futuro indicativo',
  cond: 'Condicional',
  psub: 'Presente subjentivo'
  // gppt: 'Gerundio'
  //   pisb1: 'Pretérito imperfecto',
  //   pisb2: 'Pretérito imperfecto 2',
  //   fsub: 'Futuro',
  //   ppci: 'Pretérito perfecto compuesto',
  //   ppsi: 'Pretérito pluscuamperfecto',
  //   pant: 'Pretérito anterior',
  //   fpin: 'Futuro perfecto',
  //   cpef: 'Condicional perfecto',
  //   ppfs: 'Pretérito perfecto',
  //   ppss1: 'Pretérito pluscuamperfecto',
  //   ppss2: 'Pretérito pluscuamperfecto 2',
  //   fpsb: 'Futuro perfecto',
};
