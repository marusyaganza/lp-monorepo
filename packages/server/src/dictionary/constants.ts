import { Tense } from 'generated/graphql';

export type TagType = {
  tag: string;
  replacement: string;
};

export type ComplexTagType = {
  opening: string;
  closing: string;
};

export const TAGS: TagType[] = [
  { tag: '{b}', replacement: '<b>' },
  { tag: '{/b}', replacement: '</b>' },
  { tag: '{ldquo}', replacement: '\u201C' },
  { tag: '{rdquo}', replacement: '\u201D' },
  { tag: '{it}', replacement: '<i>' },
  { tag: '{/it}', replacement: '</i>' },
  { tag: '{wi}', replacement: '<i>' },
  { tag: '{/wi}', replacement: '</i>' },
  { tag: '{gloss}=', replacement: '<i>' },
  { tag: '{/gloss}', replacement: '</i>' },
  { tag: '{sc}', replacement: '<i>' },
  { tag: '{/sc}', replacement: '</i>' },
  { tag: '{parahw}', replacement: '</i>' },
  { tag: '{/parahw}', replacement: '</i>' },
  { tag: '{phrase}', replacement: '</i>' },
  { tag: '{/phrase}', replacement: '</i>' },
  { tag: '{qword}', replacement: '</i>' },
  { tag: '{/qword}', replacement: '</i>' },
  { tag: '{itsc}', replacement: '<i>' },
  { tag: '{/itsc}', replacement: '</i>' },
  { tag: '{inf}', replacement: '' },
  { tag: '{/inf}', replacement: '' },
  { tag: '{sup}', replacement: '' },
  { tag: '{/sup}', replacement: '' },
  { tag: '{bc}', replacement: '' },
  { tag: '{dx}', replacement: '' },
  { tag: '{/dx}', replacement: '' },
  { tag: '{dx_def}', replacement: '' },
  { tag: '{/dx_def}', replacement: '' },
  { tag: '{dx_ety}', replacement: '' },
  { tag: '{/dx_ety}', replacement: '' },
  { tag: '{ma}', replacement: '' },
  { tag: '{/ma}', replacement: '' },
  { tag: '{rom}', replacement: '' },
  { tag: '{/rom}', replacement: '' }
];

export const COMPLEX_TAGS: ComplexTagType[] = [
  { opening: '{sx', closing: '}' },
  { opening: '{d_link', closing: '}' },
  { opening: '{a_link', closing: '}' },
  { opening: '{i_link', closing: '}' },
  { opening: '{et_link', closing: '}' },
  { opening: '{mat', closing: '}' },
  { opening: '{dxt', closing: '}' }
];

export const COMPLEX_TAGS_TO_REMOVE: ComplexTagType[] = [
  { opening: '{ds|', closing: '}' }
];

export const HW_TAGS: TagType[] = [
  { tag: '{bit}', replacement: '' },
  { tag: '{/bit}', replacement: '' }
];

// Uncomment more tenses as needed
export const TENSES: Record<Tense, string> = {
  pind: 'Presente indicativo',
  pprf: 'Pretérito perfecto simple',
  impf: 'Imperativo',
  pret: 'Pretérito imperfecto'
  //   futr: 'Futuro indicativo',
  //   cond: 'Condicional',
  //   psub: 'Presente subjentivo',
  //   pisb1: 'Pretérito imperfecto',
  //   pisb2: 'Pretérito imperfecto 2',
  //   fsub: 'Futuro',
  //   ppci: 'Pretérito perfecto compuesto',
  //   ppsi: 'Pretérito pluscuamperfecto',
  //   pant: 'Pretérito anterior',
  //   fpin: 'Futuro perfecto',
  //   cpef: 'Condicional perfecto',
  //   ppfs: 'Pretérito perfecto',
  //   ppss1: 'Pretérito pluscuamperfecto',
  //   ppss2: 'Pretérito pluscuamperfecto 2',
  //   fpsb: 'Futuro perfecto',
};
