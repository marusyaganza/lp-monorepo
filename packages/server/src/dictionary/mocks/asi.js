const asi = [
  {
    meta: {
      id: 'asi:1',
      uuid: 'c68c2865-eb3e-4e93-9ca1-18d9029850f2',
      lang: 'es',
      sort: '2602157000',
      src: 'spanish',
      section: 'alpha',
      stems: [
        'así',
        'así así',
        'así como',
        'así como así',
        'así nomás',
        'así de',
        'así mismo',
        'así que',
        'así y todo'
      ],
      offensive: false
    },
    hom: 1,
    hwi: {
      hw: 'así',
      prs: [
        {
          sound: {
            audio: 'asi0001sp'
          }
        }
      ]
    },
    fl: 'adverb',
    def: [
      {
        sseq: [
          [
            [
              'sense',
              {
                sn: '1',
                dt: [
                  [
                    'text',
                    '{bc}{a_link|like this}, {a_link|like that}, {a_link|so} '
                  ],
                  [
                    'vis',
                    [
                      {
                        t: 'así se hace',
                        tr: "that's how it's done"
                      },
                      {
                        t: 'no puede seguir así',
                        tr: "it can't go on like this"
                      },
                      {
                        t: 'así sea',
                        tr: 'so be it'
                      },
                      {
                        t: 'y así sucesivamente',
                        tr: 'and so on'
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '2',
                vrs: [
                  {
                    va: 'así así'
                  }
                ],
                dt: [['text', '{bc}{a_link|so-so}, {a_link|fair}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '3',
                vrs: [
                  {
                    va: 'así como',
                    vac: '~ como'
                  }
                ],
                dt: [['text', '{bc}{a_link|as well as}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '4',
                vrs: [
                  {
                    va: 'así como así'
                  },
                  {
                    vl: 'or',
                    va: 'así nomás'
                  }
                ],
                dt: [['text', '{bc}just like that']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '5',
                vrs: [
                  {
                    va: 'así de',
                    vac: '~ de'
                  }
                ],
                dt: [
                  ['text', '{bc}{a_link|so}, about so '],
                  [
                    'vis',
                    [
                      {
                        t: 'una caja así de grande',
                        tr: 'a box about so big'
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sen',
              {
                sn: '6',
                vrs: [
                  {
                    va: 'así mismo'
                  }
                ],
                xrs: [
                  [
                    {
                      xrt: 'asimismo',
                      xref: 'asimismo'
                    }
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '7',
                vrs: [
                  {
                    va: 'así que'
                  }
                ],
                dt: [['text', '{bc}{a_link|so}, {a_link|therefore}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '8',
                vrs: [
                  {
                    va: 'así y todo'
                  }
                ],
                dt: [['text', '{bc}{a_link|even so}']]
              }
            ]
          ]
        ]
      }
    ],
    shortdef: ['like this, like that, so', 'so-so, fair', 'as well as']
  },
  {
    meta: {
      id: 'asi:2',
      uuid: 'ab2d1ae3-4393-496d-bff9-e4bb5a4249f1',
      lang: 'es',
      sort: '2602158000',
      src: 'spanish',
      section: 'alpha',
      stems: ['así'],
      offensive: false
    },
    hom: 2,
    hwi: {
      hw: 'así',
      prs: [
        {
          sound: {
            audio: 'asi0002sp'
          }
        }
      ]
    },
    fl: 'adjective',
    def: [
      {
        sseq: [
          [
            [
              'sense',
              {
                dt: [
                  ['text', '{bc}{a_link|such}, such a, {a_link|like that} '],
                  [
                    'vis',
                    [
                      {
                        t: 'un talento así',
                        tr: 'a talent like that'
                      },
                      {
                        t: 'algo así',
                        tr: 'something like that'
                      },
                      {
                        t: 'así es la vida',
                        tr: "that's life"
                      },
                      {
                        t: 'si es así …',
                        tr: "if that's the case …"
                      }
                    ]
                  ]
                ]
              }
            ]
          ]
        ]
      }
    ],
    shortdef: ['such, such a, like that']
  },
  {
    meta: {
      id: 'asi:3',
      uuid: 'efad688a-6580-4ca1-b144-54f184e9342a',
      lang: 'es',
      sort: '2602159000',
      src: 'spanish',
      section: 'alpha',
      stems: ['así'],
      offensive: false
    },
    hom: 3,
    hwi: {
      hw: 'así',
      prs: [
        {
          sound: {
            audio: 'asi0003sp'
          }
        }
      ]
    },
    fl: 'conjunction',
    def: [
      {
        sseq: [
          [
            [
              'sense',
              {
                dt: [
                  [
                    'text',
                    '{sx|aunque||} {bc}{a_link|even if}, {a_link|even though} '
                  ],
                  [
                    'vis',
                    [
                      {
                        t: 'no irá, así le paguen',
                        tr: "he won't go, even if they pay him"
                      }
                    ]
                  ]
                ]
              }
            ]
          ]
        ]
      }
    ],
    shortdef: ['aunque : even if, even though']
  },
  {
    meta: {
      id: 'aun',
      uuid: 'b5d076af-9a0d-4f1f-b6e6-b4628b712524',
      lang: 'es',
      sort: '2602406000',
      src: 'spanish',
      section: 'alpha',
      stems: ['aun', 'aun así', 'aun más'],
      offensive: false
    },
    hwi: {
      hw: 'aun',
      prs: [
        {
          sound: {
            audio: 'aun0001sp'
          }
        }
      ]
    },
    fl: 'adverb',
    def: [
      {
        sseq: [
          [
            [
              'sense',
              {
                sn: '1',
                dt: [
                  ['text', '{bc}{a_link|even} '],
                  [
                    'vis',
                    [
                      {
                        t: 'ni aun en coche llegaría a tiempo',
                        tr: "I wouldn't arrive on time even if I drove"
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '2',
                vrs: [
                  {
                    va: 'aun así'
                  }
                ],
                dt: [['text', '{bc}{a_link|even so}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '3',
                vrs: [
                  {
                    va: 'aun más'
                  }
                ],
                dt: [['text', '{bc}even more']]
              }
            ]
          ]
        ]
      }
    ],
    shortdef: ['even', 'even so', 'even more']
  },
  {
    meta: {
      id: 'como:3',
      uuid: '580e2bcb-e643-44fa-8d11-81cb5db5ceb4',
      lang: 'es',
      sort: '2605203000',
      src: 'spanish',
      section: 'alpha',
      stems: ['como', 'así como'],
      offensive: false
    },
    hom: 3,
    hwi: {
      hw: 'como',
      prs: [
        {
          sound: {
            audio: 'como004sp'
          }
        }
      ]
    },
    fl: 'preposition',
    def: [
      {
        sseq: [
          [
            [
              'sense',
              {
                sn: '1',
                dt: [
                  ['text', '{bc}{a_link|like}, {a_link|as} '],
                  [
                    'vis',
                    [
                      {
                        t: 'ligero como una pluma',
                        tr: 'light as a feather'
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '2',
                vrs: [
                  {
                    va: 'así como'
                  }
                ],
                dt: [['text', '{bc}{a_link|as well as}']]
              }
            ]
          ]
        ]
      }
    ],
    shortdef: ['like, as', 'as well as']
  },
  {
    meta: {
      id: 'decir:1',
      uuid: '48a4e281-4007-423c-9bb9-05dbae10f766',
      lang: 'es',
      sort: '2606512000',
      src: 'spanish',
      section: 'alpha',
      stems: [
        'decir',
        'como quien dice',
        'es decir',
        'dicho y hecho',
        '(o) mejor dicho',
        '¡no me digas!',
        'por así decirlo',
        'querer decir',
        'decirse'
      ],
      offensive: false
    },
    hom: 1,
    hwi: {
      hw: 'decir',
      prs: [
        {
          sound: {
            audio: 'decir01sp'
          }
        }
      ]
    },
    fl: 'transitive verb',
    def: [
      {
        sseq: [
          [
            [
              'sense',
              {
                sn: '1',
                dt: [
                  ['text', '{bc}to {a_link|say} '],
                  [
                    'vis',
                    [
                      {
                        t: 'dice que no irá',
                        tr: "she says she won't go"
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '2',
                dt: [
                  ['text', '{bc}to {a_link|tell} '],
                  [
                    'vis',
                    [
                      {
                        t: 'dime lo que estás pensando',
                        tr: "tell me what you're thinking"
                      },
                      {
                        t: 'ya te lo decía yo',
                        tr: 'I told you so'
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '3',
                dt: [
                  ['text', '{bc}to {a_link|tell}, to {a_link|say} '],
                  [
                    'vis',
                    [
                      {
                        t: 'haz lo que te digo',
                        tr: 'do as I say, do what I tell you'
                      },
                      {
                        t: 'te dije que callaras',
                        tr: 'I told you to be quiet'
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '4',
                dt: [
                  ['text', '{bc}to {a_link|speak}, to {a_link|talk} '],
                  [
                    'vis',
                    [
                      {
                        t: 'no digas tonterías',
                        tr: "don't talk nonsense"
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '5',
                dt: [
                  ['text', '{bc}to {a_link|call} '],
                  [
                    'vis',
                    [
                      {
                        t: 'me dicen Rosy',
                        tr: 'they call me Rosy'
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '6',
                vrs: [
                  {
                    va: 'como quien dice'
                  }
                ],
                dt: [['text', '{bc}so to speak']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '7',
                vrs: [
                  {
                    va: 'es decir'
                  }
                ],
                dt: [['text', '{bc}{a_link|that is to say}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '8',
                vrs: [
                  {
                    va: 'dicho y hecho'
                  }
                ],
                dt: [['text', '{bc}{a_link|no sooner said than done}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '9',
                vrs: [
                  {
                    va: '(o) mejor dicho'
                  }
                ],
                dt: [['text', '{bc}(or) {a_link|rather}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '10',
                vrs: [
                  {
                    va: '¡no me digas!'
                  }
                ],
                dt: [['text', "{bc}you're kidding!, {a_link|you don't say!}"]]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '11',
                vrs: [
                  {
                    va: 'por así decirlo'
                  }
                ],
                dt: [['text', '{bc}so to speak']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '12',
                vrs: [
                  {
                    va: 'querer decir'
                  }
                ],
                dt: [
                  ['text', '{bc}to {a_link|mean} '],
                  [
                    'vis',
                    [
                      {
                        t: '¿qué quiere decir?',
                        tr: 'what do you mean?'
                      }
                    ]
                  ]
                ]
              }
            ]
          ]
        ]
      }
    ],
    dros: [
      {
        drp: 'decirse',
        fl: 'reflexive verb',
        def: [
          {
            sseq: [
              [
                [
                  'sense',
                  {
                    sn: '1',
                    dt: [['text', '{bc}to say to oneself']]
                  }
                ]
              ],
              [
                [
                  'sense',
                  {
                    sn: '2',
                    dt: [
                      ['text', '{bc}to be said '],
                      [
                        'vis',
                        [
                          {
                            t: '¿cómo se dice {ldquo}lápiz{rdquo} en francés?',
                            tr: 'how do you say {ldquo}pencil{rdquo} in French?'
                          }
                        ]
                      ]
                    ]
                  }
                ]
              ]
            ]
          }
        ]
      }
    ],
    suppl: {
      cjts: [
        {
          cjid: 'gppt',
          cjfs: ['diciendo', 'dicho']
        },
        {
          cjid: 'pind',
          cjfs: ['digo', 'dices', 'dice', 'decimos', 'decís', 'dicen']
        },
        {
          cjid: 'pret',
          cjfs: ['decía', 'decías', 'decía', 'decíamos', 'decíais', 'decían']
        },
        {
          cjid: 'pprf',
          cjfs: ['dije', 'dijiste', 'dijo', 'dijimos', 'dijisteis', 'dijeron']
        },
        {
          cjid: 'futr',
          cjfs: ['diré', 'dirás', 'dirá', 'diremos', 'diréis', 'dirán']
        },
        {
          cjid: 'cond',
          cjfs: ['diría', 'dirías', 'diría', 'diríamos', 'diríais', 'dirían']
        },
        {
          cjid: 'psub',
          cjfs: ['diga', 'digas', 'diga', 'digamos', 'digáis', 'digan']
        },
        {
          cjid: 'pisb1',
          cjfs: [
            'dijera',
            'dijeras',
            'dijera',
            'dijéramos',
            'dijerais',
            'dijeran'
          ]
        },
        {
          cjid: 'pisb2',
          cjfs: [
            'dijese',
            'dijeses',
            'dijese',
            'dijésemos',
            'dijeseis',
            'dijesen'
          ]
        },
        {
          cjid: 'fsub',
          cjfs: [
            'dijere',
            'dijeres',
            'dijere',
            'dijéremos',
            'dijereis',
            'dijeren'
          ]
        },
        {
          cjid: 'ppci',
          cjfs: [
            'he dicho',
            'has dicho',
            'ha dicho',
            'hemos dicho',
            'habéis dicho',
            'han dicho'
          ]
        },
        {
          cjid: 'ppsi',
          cjfs: [
            'había dicho',
            'habías dicho',
            'había dicho',
            'habíamos dicho',
            'habíais dicho',
            'habían dicho'
          ]
        },
        {
          cjid: 'pant',
          cjfs: [
            'hube dicho',
            'hubiste dicho',
            'hubo dicho',
            'hubimos dicho',
            'hubisteis dicho',
            'hubieron dicho'
          ]
        },
        {
          cjid: 'fpin',
          cjfs: [
            'habré dicho',
            'habrás dicho',
            'habrá dicho',
            'habremos dicho',
            'habréis dicho',
            'habrán dicho'
          ]
        },
        {
          cjid: 'cpef',
          cjfs: [
            'habría dicho',
            'habrías dicho',
            'habría dicho',
            'habríamos dicho',
            'habríais dicho',
            'habrían dicho'
          ]
        },
        {
          cjid: 'ppfs',
          cjfs: [
            'haya dicho',
            'hayas dicho',
            'haya dicho',
            'hayamos dicho',
            'hayáis dicho',
            'hayan dicho'
          ]
        },
        {
          cjid: 'ppss1',
          cjfs: [
            'hubiera dicho',
            'hubieras dicho',
            'hubiera dicho',
            'hubiéramos dicho',
            'hubierais dicho',
            'hubieran dicho'
          ]
        },
        {
          cjid: 'ppss2',
          cjfs: [
            'hubiese dicho',
            'hubieses dicho',
            'hubiese dicho',
            'hubiésemos dicho',
            'hubieseis dicho',
            'hubiesen dicho'
          ]
        },
        {
          cjid: 'fpsb',
          cjfs: [
            'hubiere dicho',
            'hubieres dicho',
            'hubiere dicho',
            'hubiéremos dicho',
            'hubiereis dicho',
            'hubieren dicho'
          ]
        },
        {
          cjid: 'impf',
          cjfs: ['-', 'di', 'diga', 'digamos', 'decid', 'digan']
        }
      ]
    },
    shortdef: ['to say', 'to tell', 'to tell, to say']
  },
  {
    meta: {
      id: 'tan:2',
      uuid: '6ad0961a-7c11-43ab-ad80-5e3a94c8e116',
      lang: 'es',
      sort: '2621448000',
      src: 'spanish',
      section: 'alpha',
      stems: ['tan', 'tan es así'],
      offensive: false
    },
    hom: 2,
    hwi: {
      hw: 'tan'
    },
    fl: 'pronoun',
    def: [
      {
        sseq: [
          [
            [
              'sense',
              {
                vrs: [
                  {
                    va: 'tan es así'
                  }
                ],
                dt: [['text', '{bc}so much so']]
              }
            ]
          ]
        ]
      }
    ],
    shortdef: ['so much so']
  },
  {
    meta: {
      id: 'tanto:4',
      uuid: 'f6712a94-2c57-4508-851f-89a9f6ab81d9',
      lang: 'es',
      sort: '2621463000',
      src: 'spanish',
      section: 'alpha',
      stems: [
        'tanto',
        'tanta',
        'en tanto',
        'entre tanto',
        'otro tanto',
        'por lo tanto',
        'tanto es así'
      ],
      offensive: false
    },
    hom: 4,
    hwi: {
      hw: 'tanto',
      prs: [
        {
          sound: {
            audio: 'tanto04sp'
          }
        }
      ]
    },
    ahws: [
      {
        hw: 'tanta',
        hwc: '-ta'
      }
    ],
    fl: 'pronoun',
    def: [
      {
        sseq: [
          [
            [
              'sense',
              {
                sn: '1',
                dt: [
                  ['text', '{bc}so much, so many '],
                  [
                    'vis',
                    [
                      {
                        t: 'tiene tanto que hacer',
                        tr: 'she has so much to do'
                      },
                      {
                        t: '¡no me des tantos!',
                        tr: "don't give me so many!"
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '2',
                vrs: [
                  {
                    va: 'en tanto',
                    vac: 'en ~'
                  }
                ],
                dt: [['text', '{bc}{a_link|while}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '3',
                vrs: [
                  {
                    va: 'entre tanto',
                    vac: 'entre ~'
                  }
                ],
                dt: [['text', '{bc}{a_link|meanwhile}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '4',
                vrs: [
                  {
                    va: 'otro tanto'
                  }
                ],
                dt: [
                  ['text', '{bc}again as much, again as many '],
                  [
                    'vis',
                    [
                      {
                        t: 'tiene un metro de ancho y otro tanto de altura',
                        tr: "it's a meter wide and a meter high"
                      },
                      {
                        t: 'otro tanto podría decirse de …',
                        tr: 'the same can be said of …'
                      }
                    ]
                  ]
                ]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '5',
                vrs: [
                  {
                    va: 'por lo tanto'
                  }
                ],
                dt: [['text', '{bc}{a_link|therefore}']]
              }
            ]
          ],
          [
            [
              'sense',
              {
                sn: '6',
                vrs: [
                  {
                    va: 'tanto es así'
                  }
                ],
                dt: [['text', '{bc}so much so']]
              }
            ]
          ]
        ]
      }
    ],
    shortdef: ['so much, so many', 'while', 'meanwhile']
  }
];

export default asi;
