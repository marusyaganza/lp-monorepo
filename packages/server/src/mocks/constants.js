export const mockEnv = {
  SALT_NUMBER: 3,
  JWT_SECTET: 'JWT_secret'
};
