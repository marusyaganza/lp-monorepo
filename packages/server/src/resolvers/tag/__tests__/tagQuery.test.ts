import {
  createTestServer,
  getDataFromGQL,
  getErrorMessageFromGQL
} from '../../../tests/helpers';
import { models } from '../../../tests/mocks/mockModels';
import { user } from '../../../tests/mocks/data';
import { tagsQueries } from '../../../tests/mocks/gqlQueries';
import { Language } from '../../../generated/graphql';
import { ERROR_MESSAGES } from '../../../constants/errorMessages';

describe('tags queries', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.resetAllMocks();
  });

  test('tags, English', async () => {
    const { query } = createTestServer({
      user,
      models
    });

    const res = await query({
      query: tagsQueries.tagsQuery,
      variables: { language: Language.English }
    });
    expect(getDataFromGQL(res)).toMatchSnapshot();
    expect(getErrorMessageFromGQL(res)).toEqual(undefined);
  });

  test('tags, Spanish', async () => {
    const { query } = createTestServer({
      user,
      models
    });

    const res = await query({
      query: tagsQueries.tagsQuery,
      variables: { language: Language.Spanish }
    });
    expect(getDataFromGQL(res)).toMatchSnapshot();
    expect(getErrorMessageFromGQL(res)).toEqual(undefined);
  });

  test('tags if user is undefined', async () => {
    const { query } = createTestServer({
      models: {
        ...models,
        User: {
          findOne: jest.fn()
        }
      }
    });

    const res = await query({
      query: tagsQueries.tagsQuery,
      variables: { language: Language.English }
    });
    expect(getErrorMessageFromGQL(res)).toEqual(
      ERROR_MESSAGES.NOT_AUTHENTICATED
    );
  });

  test('tags if user is not found', async () => {
    const { query } = createTestServer({
      user,
      models: {
        ...models,
        User: {
          findOne: jest.fn()
        }
      }
    });

    const res = await query({
      query: tagsQueries.tagsQuery,
      variables: { language: Language.English }
    });
    expect(getErrorMessageFromGQL(res)).toEqual(
      ERROR_MESSAGES.NOT_AUTHENTICATED
    );
  });
});
