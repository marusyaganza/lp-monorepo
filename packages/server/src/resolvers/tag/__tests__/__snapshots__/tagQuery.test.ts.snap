// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tags queries tags, English 1`] = `
{
  "tags": [
    {
      "color": "#d1af33",
      "desc": "Tag 1 desc",
      "id": "tag1Id",
      "text": "Spanish Tag 1 text",
    },
    {
      "color": "#ee7777",
      "desc": "Tag 2 desc",
      "id": "tag2Id",
      "text": "English Tag 2 text",
    },
  ],
}
`;

exports[`tags queries tags, Spanish 1`] = `
{
  "tags": [
    {
      "color": "#d1af33",
      "desc": "Tag 1 desc",
      "id": "tag1Id",
      "text": "Spanish Tag 1 text",
    },
    {
      "color": "#ee7777",
      "desc": "Tag 2 desc",
      "id": "tag2Id",
      "text": "English Tag 2 text",
    },
  ],
}
`;
